src/createServer.js

```diff
@@ -1,3 +1,50 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { validateRequest } = require('./validateRequest');
+const { generateRespond } = require('./generateRespond');
+
+function createServer() {
+  const faviconURL = '/favicon.ico';
+
+  const server = http.createServer((req, res) => {
+    try {
+      const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+      if (normalizedUrl.pathname === faviconURL) {
+        res.end();
+
+        return;
+      }
+
+      const textToTransform = normalizedUrl.pathname.slice(1);
+      const toCaseTransform = normalizedUrl.searchParams.get('toCase');
+      const isValidData = validateRequest(textToTransform, toCaseTransform);
+
+      if (!isValidData.isValid) {
+        const error = { errors: isValidData.errorArray };
+
+        res.writeHead(400, 'Bad reguest', {
+          'content-type': 'application/json',
+        });
+        res.write(JSON.stringify(error));
+        res.end();
```

Comment:
You can provide a response in the `end()` method

```suggestion
        res.end(JSON.stringify(error));
```